### ISSUE TITLE
Feature Request: More padding mode.

### ISSUE THREAD

**Issue by Soptq** [Xilinx/brevitas#994]
In many applications we want to make sure that the padding alters the local feature structure and global statistics the least (e.g. Image-to-image translation). Using zero padding in these applications will result in peaking edges[1][2].

However, only zero padding is supported by brevitas by now. I wonder is it possible to also support other padding mode like reflection or symmetric padding in brevitas? or is there a workaround?

[1] https://arxiv.org/abs/1703.10593
[2] https://arxiv.org/abs/1811.11718

**Comment by Giuseppe5** [comment]
I think it should be relatively straight forward to implement your own padding mode.
Inherit a new QuantConv from brevitas' one, then you'd need to override the init and this:
https://github.com/Xilinx/brevitas/blob/200456825f3b4b8db414f2b25b64311f82d3991a/src/brevitas/nn/quant_conv.py#L200-L205

in order to call your custom padding mode. As long as you call torch.nn.functional.conv2d (or 1d, etc.) with the input/weight/bias tensor that are passed to `inner_forward_impl`, everything else with respect to quantization is handled automatically.

**Comment by Giuseppe5** [comment]
If you think that this feature could be beneficial for others and are willing to contribute to Brevitas, we would be more than happy to accept a PR.

**Comment by Soptq** [comment]
Sure, I will try to make a PR later ;)

**Comment by Giuseppe5** [comment]
Closing this now. Feel free to re-open if you need more help!
